src/llmq/quorums_instantsend.cpp

```diff
@@ -384,18 +384,6 @@ bool CInstantSendManager::ProcessTx(const CTransaction& tx, bool allowReSigning,
         return true;
     }
 
-    // In case the islock was received before the TX, filtered announcement might have missed this islock because
-    // we were unable to check for filter matches deep inside the TX. Now we have the TX, so we should retry.
-    uint256 islockHash;
-    {
-        LOCK(cs);
-        islockHash = db.GetInstantSendLockHashByTxid(tx.GetHash());
-    }
-    if (!islockHash.IsNull()) {
-        CInv inv(MSG_ISLOCK, islockHash);
-        g_connman->RelayInvFiltered(inv, tx, LLMQS_PROTO_VERSION);
```

Comment:
Good find! But I think this logic should be moved (at least) to `TransactionAddedToMempool`, not dropped completely.

Reply:
Why is this?

Reply:
This explanation `In case the islock was received before the TX, filtered announcement might have missed this islock because we were unable to check for filter matches deep inside the TX. Now we have the TX, so we should retry.` makes sense to me. The bug is not only that it's a wasted db call (like you pointed out), it's also a bug that this `RelayInvFiltered` logic is never executed.